soda-tests/tests/features/test_soda_data_types.py

```diff
@@ -10,3 +17,137 @@ def test_soda_data_types(data_source_test_helper: DataSourceTestHelper):
+        .column_varchar("varchar_default")
+        .column_varchar("varchar_w_length", 100)
+        .column_text("text_default")
+        .column_smallint("smallint_default")
+        .column_integer("integer_default")
+        .column_bigint("bigint_default")
+        .column_decimal("decimal_default")
+        .column_decimal("decimal_w_precision", 10)
+        .column_decimal("decimal_w_precision_and_scale", 10, 2)
+        .column_numeric("numeric_default")
+        .column_numeric("numeric_w_precision", 10)
+        .column_numeric("numeric_w_precision_and_scale", 10, 2)
+        .column_float("float_default")
+        .column_double("double_default")
+        .column_timestamp("timestamp_default")
+        .column_timestamp("timestamp_w_precision", 2)
+        .column_timestamp_tz("timestamp_tz_default")
+        .column_timestamp_tz("timestamp_tz_w_precision", 4)
+        .column_date("date_default")
+        .column_time("time_default")
+        .column_boolean("boolean_default")
+        .rows(
+            rows=[
+                (
+                    # Chars
+                    "a",
+                    "abc",
+                    # Varchars
+                    "a",
+                    "abc",
+                    # Texts
+                    "a",
+                    # Integers
+                    1,
+                    100,
+                    100000,
+                    # Decimals
+                    1.0,
+                    100.0,
+                    100000.1234567890,
+                    # Numerics
+                    1.0,
+                    100.0,
+                    100000.1234567890,
+                    # Floats
+                    1.1234,
+                    # Doubles
+                    1.1234567890,
+                    # Timestamps
+                    datetime.datetime(2021, 1, 1, 1, 1, 1),
+                    datetime.datetime(2022, 1, 1, 1, 1, 1, 12),
+                    # Timestamp Zones
+                    datetime.datetime(2023, 1, 1, 1, 1, 1),
+                    datetime.datetime(2024, 1, 1, 1, 1, 1, 12),
+                    # Dates
+                    datetime.date(2025, 1, 1),
+                    # Times
+                    datetime.time(1, 2, 3),
+                    # Booleans
+                    True,
+                )
+            ]
+        )
+        .build()
+    )
+
+    # Building the test table already tests the mapping from canonical to data type
+    # Behind the scenes, we build this table using SodaDataTypeName.CHAR, SodaDataTypeName.SMALLINT, etc.
+    # So if the table gets created, then the mapping from canonical to data type is correct
+    test_table = data_source_test_helper.ensure_test_table(test_table_specification)
+
+    actual_columns: list[ColumnMetadata] = data_source_test_helper.data_source_impl.get_columns_metadata(
+        dataset_prefixes=test_table.dataset_prefix, dataset_name=test_table.unique_name
+    )
+
+    assert len(actual_columns) == 23
```

Comment:
I'm sure we'll change the list of columns at some point 😅

```suggestion
    assert len(actual_columns) == len(test_table_specification.columns)
```